### PR TITLE
Coverity fixes - memory leaks

### PR DESCRIPTION
**Pull request by XVilka**
# SQUASH ME

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Just a few minor memory leaks

**Test plan**

CI is green

### CODE REVIEW

File: librz/core/cautocmpl.c [hunk 1]
```diff
@@ -271,6 +271,7 @@ static void autocmplt_cmd_arg_file(RzLineNSCompletionResult *res, const char *s,
 #endif
 		char *tmp = rz_str_newf(fmt, RZ_SYS_DIR, input);
 		if (!tmp) {
+			free(input);
```
Comment (lkabuci) [inline librz/core/cautocmpl.c]: this isn't necessary, if it's NULL then it's not allocated

**Review by lkabuci — APPROVED** [review]
Looks good to me.

File: librz/core/cautocmpl.c (hunk 1 above)
Comment (wargio) [reply librz/core/cautocmpl.c]: free (and all the free method in rizin) accepts NULL pointers.

File: librz/core/cautocmpl.c [hunk 2]
```diff
@@ -271,6 +271,7 @@ static void autocmplt_cmd_arg_file(RzLineNSCompletionResult *res, const char *s,
...
 			return;
 		}
 		free(input);
```
Comment (wargio) [inline librz/core/cautocmpl.c]:
```suggestion
		char *tmp = rz_str_newf(fmt, RZ_SYS_DIR, input);
		free(input);
		if (!tmp) {
			return;
		}
```
input is always freed, so we can free it after `rz_str_newf`

**Review by wargio — CHANGES_REQUESTED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**